### PR TITLE
Add schema directives validation

### PR DESCRIPTION
**Pull request by javimartinez**
Following the schema validation rules for directives posted in #232, I've implemented the 3 and 4.1 rules in this PR. 

I think that due to the type model, it isn't necessary to check the rest of the rules, but I'm no pretty sure, am I missing something?

### CODE REVIEW

File: core/src/main/scala/caliban/validation/Validator.scala [hunk 1]
```diff
@@ -722,6 +728,45 @@ object Validator {
...
+      errorContext: String
+    ): IO[ValidationError, Unit] =
+      IO.foreach_(directivesExtractor(t).getOrElse(List.empty))(validateDirective(_, errorContext))
+
+    IO.foreach_(types) { t =>
+      val typeErrorContext                    = s"Type '${t.name.getOrElse("")}'"
+      val fieldErrorContext: String => String = fieldName => s"Field '$fieldName' of $typeErrorContext"
+      for {
+        _ <- validateDirectives[__Type](t, _.directives, typeErrorContext)
+        _ <- IO.foreach_(t.fields(__DeprecatedArgs(Some(true))).getOrElse(List.empty[__Field]))(f =>
```
Comment (ghostdogpr) [inline core/src/main/scala/caliban/validation/Validator.scala:762]: There can also be directives in `__Type.inputFields` and `__Field.args`: both contain `__InputValue` which has `directives`.

Comment (javimartinez) [reply core/src/main/scala/caliban/validation/Validator.scala]: I haven't seen those, added!

File: core/src/main/scala/caliban/validation/Validator.scala [hunk 2]
```diff
@@ -722,6 +728,65 @@ object Validator {
...
+    def validateDirective(directive: Directive, errorContext: String) = {
+      val directiveErrorContext = s"Directive '${directive.name}' of $errorContext"
+      for {
+        _ <- doesNotStartWithUnderscore(directive, directiveErrorContext)
+        _ <- validateArguments(directive.arguments, directiveErrorContext)
+      } yield ()
+    }
+
+    def validateDirectives[T](
+      t: T,
```
Comment (ghostdogpr) [inline core/src/main/scala/caliban/validation/Validator.scala:751]: You pass `T` and `T => Option[List[Directive]]`, why not just pass `Option[List[Directive]]`? Then you won't need the type parameter anymore.

Comment (javimartinez) [reply core/src/main/scala/caliban/validation/Validator.scala]: done :)

**Pull request merged**